internal/k6runner/local.go

```diff
@@ -115,36 +120,68 @@ func (r Local) Run(ctx context.Context, script Script) (*RunResponse, error) {
 	cmd.Stderr = &stderr
 
 	start := time.Now()
+	logger.Info().Str("command", cmd.String()).Bytes("script", script.Script).Msg("running k6 script")
+	err = cmd.Run()
 
-	r.logger.Info().Str("command", cmd.String()).Bytes("script", script.Script).Msg("running k6 script")
+	duration := time.Since(start)
 
-	if err := cmd.Run(); err != nil {
-		r.logger.Error().Err(err).Str("stdout", stdout.String()).Str("stderr", stderr.String()).Msg("k6 exited with error")
-		return nil, fmt.Errorf("cannot execute k6 script: %w", err)
-	}
+	// If context error is non-nil, incorporate it into err.
+	// This brings context to log lines and plays well with both errors.Is and errors.As.
+	err = errors.Join(err, ctx.Err())
 
-	duration := time.Since(start)
+	if err != nil && !isUserError(err) {
+		logger.Error().
+			Err(err).
+			Dur("duration", duration).
+			Msg("cannot run k6")
 
-	r.logger.Debug().Str("stdout", stdout.String()).Str("stderr", stderr.String()).Dur("duration", duration).Msg("ran k6 script")
-	r.logger.Info().Dur("duration", duration).Msg("ran k6 script")
+		dumpK6OutputStream(r.logger, zerolog.ErrorLevel, &stdout, "stream", "stdout")
+		dumpK6OutputStream(r.logger, zerolog.ErrorLevel, &stderr, "stream", "stderr")
 
-	var result RunResponse
+		logs, _ := afs.ReadFile(logsFn)
+		dumpK6OutputStream(r.logger, zerolog.InfoLevel, bytes.NewReader(logs), "stream", "logs")
```

Comment:
_nit:_ I understand these dumps are just for our own observability on the k6 output when the test is errored by a non-user / non-k6 error, right? Would it make sense to also log the possible error on reading the logs file and avoid the extra call to `dumpK6OutputStream` in that case?

I guess that is rare and would have low impact, but not sure if we could miss it at some point. It does feel a little bit cumbersome so up to you. Could be something like:
```suggestion
		if logs, err := afs.ReadFile(logsFn); err == nil {
			dumpK6OutputStream(r.logger, zerolog.InfoLevel, bytes.NewReader(logs), "stream", "logs")
		} else {
			logger.Error().Err(err).Msg("error reading k6 logs output")
		}
```